### PR TITLE
Update RTFS package, cleanup API calls

### PR DESCRIPTION
**Pull request by postables**
Wait until https://github.com/RTradeLtd/rtfs/pull/7 is merged, then create a new release for `rtfs` and update the version used by this PR, before merging.

## :construction_worker: Purpose
Needed to update the `rtfs` version we use
Cleaned up API calls
Better handling of auth tokens
Small amount of additional security measures
[Postman documentation has been updated](https://documenter.getpostman.com/view/4295780/RWEcQM6W)
## :rocket: Changes
`rtfs` was updated to include `krab` and `beam`

API calls were a little messy for IPFS usage, and they have been adjusted

Currently whenever we go to retrieve usernames from auth tokens, it can be possible for empty auth tokens to be provided. This is catched by our other protections, however by not checking for this, we potentially open up security holes. If we can't recover a username from the auth token, then we abort processing their request and return an error message

Small amount of additional security measures were added to the API, mostly related to XSS filtering, and cache control.

## :warning: Breaking Changes
Yes, basically every IPFS api call is renamed

### CODE REVIEW

File: api/api.go [hunk 1]
```diff
@@ -209,12 +216,34 @@ func (api *API) ListenAndServe(addr string, tls *TLSConfig) error {
...
+			log.Fatal(err)
+		}
+	}
 	// set up defaults
 	api.r.Use(
+		// slightly more complex xss removal
 		xssMdlwr.RemoveXss(),
-		limit.MaxAllowed(20),
+		// rate limiting
+		limit.MaxAllowed(connLimit),
```
Comment (bobheadxi) [inline api/api.go]: 👍 configuration injection!

File: api/routes_mini.go [hunk 2]
```diff
@@ -11,7 +11,11 @@ import (
 
 // MakeBucket is used to create a bucket in our minio container
 func (api *API) makeBucket(c *gin.Context) {
-	username := GetAuthenticatedUserFromContext(c)
+	username, err := GetAuthenticatedUserFromContext(c)
+	if err != nil {
+		api.LogError(err, eh.NoAPITokenError)(c, http.StatusBadRequest)
```
Comment (bobheadxi) [inline api/routes_mini.go]: shouldn't JWT middleware prevent unauthenticated API requests from getting through?

File: api/routes_rtfsp.go [hunk 3]
```diff
@@ -779,16 +828,99 @@ func (api *API) getUploadsByNetworkName(c *gin.Context) {
...
+		api.LogError(err, eh.NoAPITokenError)(c, http.StatusBadRequest)
+		return
+	}
+	forms := api.extractPostForms(c, "source_network", "destination_network", "content_hash")
+	if len(forms) == 0 {
+		return
+	}
+	var (
+		net1URL string
+		net2URL string
```
Comment (bobheadxi) [inline api/routes_rtfsp.go]: why not `source` and `dest`?

Comment (bobheadxi) [reply api/routes_rtfsp.go]: I think you can also declare this as:

```go
var source, dest string
```

File: api/routes_mini.go (hunk 2 above)
Comment (postables) [reply api/routes_mini.go]: That's a good point :ok_hand:

File: api/api.go [hunk 4]
```diff
@@ -209,12 +216,34 @@ func (api *API) ListenAndServe(addr string, tls *TLSConfig) error {
...
+		connLimit int
+		err       error
+	)
+	if api.cfg.API.Connection.Limit == "" {
+		connLimit = 50
+	} else {
+		// setup the connection limit
+		connLimit, err = strconv.Atoi(api.cfg.API.Connection.Limit)
+		if err != nil {
+			log.Fatal(err)
```
Comment (bobheadxi) [inline api/api.go]: we might want to have this function return err instead of panicking

File: api/routes_utils.go [hunk 5]
```diff
@@ -1,12 +1,95 @@
...
+
+// BeamContent is used to beam content from one network to another
+func (api *API) beamContent(c *gin.Context) {
+	username, err := GetAuthenticatedUserFromContext(c)
+	if err != nil {
+		api.LogError(err, eh.NoAPITokenError)(c, http.StatusBadRequest)
+		return
+	}
+	forms := api.extractPostForms(c, "source_network", "destination_network", "content_hash")
+	if len(forms) == 0 {
```
Comment (bobheadxi) [inline api/routes_utils.go]: this returns `nil` if something is missing - `len(forms)` will panic

Comment (bobheadxi) [reply api/routes_utils.go]: oops i stand corrected

**Pull request merged**